rfc/rfc-0000-match-params.md

```diff
@@ -0,0 +1,138 @@
+# Meta
+[meta]: #meta
+- Name: Match params in field selector
+- Start Date: 2022-02-14
+- Author(s): @jwntrs
+- Status: Draft
+- RFC Pull Request:
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+The [template switching RFC](https://github.com/vmware-tanzu/cartographer/pull/75) introduced the ability to match templates based on fields in the workload. However matching on params is not well defined. It is currently not possible to match on a field in the supply chain, and matching on fields in the workload is complex. You would need to define something as follows:
+
+```yaml
+  selector:
+    matchFields:
+      - key: workload.spec.params[name == promotion].value    #< ===== does this even work?
```

Comment:
I think it would look like this:
`workload.spec.params[?(@.name=="promotion")].value`

Reply:
thanks updated!